tests/src/Unit/RulesLoggerChannelTest.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\Core\Logger\LoggerChannelTest.
+ */
+
+namespace Drupal\Tests\Core\Logger;
+
+use Drupal\rules\Logger\RulesLoggerChannel;
+use Drupal\Tests\UnitTestCase;
+
+/**
+ * @coversDefaultClass \Drupal\Core\Logger\LoggerChannel
+ * @group Logger
+ */
+class RulesLoggerChannelTest extends UnitTestCase {
+
+  /**
+   * Tests LoggerChannel::log().
+   *
+   * @param int $debug_log
+   *   Is logging enabled.
+   * @param int $log_errors
+   *   Which level of messages to log.
+   *
+   * @dataProvider providerTestLog
+   * @covers ::log
+   * @covers ::setCurrentUser
+   * @covers ::setRequestStack
+   */
+  public function testLog($message_level = 0, $debug_log = 1, $log_errors = 0, $expect_log = 1) {
+    $config = $this->getConfigFactoryStub(array(
+      'rules.settings' => array(
+        'debug_log' => $debug_log,
+        'log_errors' => $log_errors,
+      )
+    ));
+    $channel = new RulesLoggerChannel($config);
+    $message = $this->randomMachineName();
+    $logger = $this->getMock('Psr\Log\LoggerInterface');
+    $channel->addLogger($logger);
+
+    $logger->expects($this->exactly($expect_log))
+      ->method('log')
+      ->with($this->anything(), $message);
```

Comment:
The first value is the message level, right? So that should be used here as well instead of this->anything()